plugins/discovery-ec2/src/main/java/org/elasticsearch/cloud/aws/AwsEc2ServiceImpl.java

```diff
@@ -125,6 +125,7 @@ public long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
             10,
             false);
         clientConfiguration.setRetryPolicy(retryPolicy);
+        clientConfiguration.setSocketTimeout((int) CLOUD_EC2.READ_TIMEOUT.get(settings).millis());
```

Comment:
I know that @tlrx prefers when we use the AWS SDK defaults instead of fixing ourselves a setting. And I agree.

Can we write instead something like:

```java
if (READ_TIMEOUT.exists(settings) || CLOUD_EC2.READ_TIMEOUT.exists(settings)) {
    clientConfiguration.setSocketTimeout((int) CLOUD_EC2.READ_TIMEOUT.get(settings).millis());
}
```

And then remove in doc that there is a default value for this setting?